### PR TITLE
Migration to AndroidX, target API 28, update kotlin to 1.2.71, support gradle plugin 3.2 and update gradle to 4.10.2

### PR DESCRIPTION
**Pull request by TimoPtr**

### CODE REVIEW

**Comment by TimoPtr** [comment]
I take the liberty to increase the version to 2.2.0

File: build.gradle [hunk 1]
```diff
@@ -2,22 +2,21 @@
...
-    ext.buildToolsVersion = '27.0.3'
+    ext.kotlin_version = '1.2.71'
+    ext.android_plugin_version = '3.2.0'
+    ext.targetSdkVersion = 28
+    ext.compileSdkVersion = 28
     ext.minSdkVersion = 14
     ext.novodaBintrayVersion = '0.8.1'
-    ext.supportLibVersion = '27.1.1'
-    ext.databinding_version = '3.1.1'
+    ext.androidX = "1.0.0"
```
Comment (ovitrif) [inline build.gradle]: I'd suggest keeping consistency for naming version variables here :)
That would mean to:
- Keep the `version` suffix
- Keep snake case (?)
- Use single quotes for value

**Comment by TimoPtr** [comment]
I fix it, I hesitate to migrate the whole things to kotlin DSL and buildSrc it could be nice I think

**Comment by bernaferrari** [comment]
It makes no sense the androidx_version variable:

    implementation "androidx.appcompat:appcompat:$androidx_version"
    implementation "com.google.android.material:material:$androidx_version"
    implementation "androidx.cardview:cardview:$androidx_version"
    implementation "androidx.vectordrawable:vectordrawable:$androidx_version"
    implementation "androidx.vectordrawable:vectordrawable-animated:$androidx_version"

The basic promise of androidx is that each library will be independently updated. cardView might be on 2.0 while vectorDrawable is at 1.0.

**Comment by ovitrif** [comment]
@bernaferrari I still think it's a decent way to keep versions in check for a library. Could serve as a good reference of the library's update status

**Comment by bernaferrari** [comment]
There is currently no problem, but in the future all versions will be fragmented.

**Comment by lisawray** [comment]
This looks great, I'll just take out the androidx-version and merge it

**Pull request merged**